pkgs/applications/misc/blender/default.nix

```diff
@@ -1,170 +1,159 @@
-{ config, stdenv, lib, fetchurl, fetchzip, boost, cmake, ffmpeg, gettext, glew
-, ilmbase, libXi, libX11, libXext, libXrender
-, libjpeg, libpng, libsamplerate, libsndfile
-, libtiff, libGLU, libGL, openal, opencolorio, openexr, openimagedenoise, openimageio2, openjpeg, python39Packages
-, openvdb, libXxf86vm, tbb, alembic
-, zlib, fftw, opensubdiv, freetype, jemalloc, ocl-icd, addOpenGLRunpath
-, jackaudioSupport ? false, libjack2
-, cudaSupport ? config.cudaSupport or false, cudatoolkit
-, colladaSupport ? true, opencollada
-, spaceNavSupport ? stdenv.isLinux, libspnav
-, makeWrapper
-, pugixml, llvmPackages, SDL, Cocoa, CoreGraphics, ForceFeedback, OpenAL, OpenGL
-, potrace
-, openxr-loader
-, embree, gmp, libharu
+# Common:
+{ stdenv, lib, config, fetchzip, fetchurl, cmake, makeWrapper
+, alembic, boost, embree, ffmpeg, fftw, freetype, gettext
+, glew, gmp, ilmbase, jemalloc, libharu, libjpeg, libpng
+, libsamplerate, libsndfile, libtiff, opencolorio, openexr
+, openimagedenoise, openimageio2, openjpeg, opensubdiv
+, potrace, pugixml, python3, tbb, zlib, zstd, runCommandNoCC
+
+# Linux specific:
+, libGL, libGLU, libX11, libXext, libXi, libXrender, libXxf86vm
+, openxr-loader, openvdb, openal
+
+# Darwin specific:
+, Cocoa, CoreGraphics, ForceFeedback, llvmPackages, OpenAL, OpenGL, SDL
+
+# Optional:
+, jackaudioSupport ? false
+, libjack2
+
+, cudaSupport ? config.cudaSupport or false
+, cudatoolkit_11, addOpenGLRunpath
+
+, colladaSupport ? true
+, opencollada
+
+, spaceNavSupport ? stdenv.isLinux
+, libspnav
+
+, waylandSupport ? false
```

Comment:
Is wayland support mutually exclusive with X11 support? if not, I'd rather default this to `stdenv.isLinux`

Reply:
From @kanashimia's comment (https://github.com/NixOS/nixpkgs/pull/148550#issuecomment-986272008), wayland support is quite experimental and disabled by default upstream.